homeassistant/components/owlet/sensor.py

```diff
@@ -0,0 +1,186 @@
+        icon="mdi:leaf",
+    ),
+    OwletSensorEntityDescription(
+        key="heartrate",
+        translation_key="heartrate",
+        native_unit_of_measurement="bpm",
+        state_class=SensorStateClass.MEASUREMENT,
+        element="heart_rate",
+        icon="mdi:heart-pulse",
+    ),
+    OwletSensorEntityDescription(
+        key="batteryminutes",
+        translation_key="batterymin",
+        native_unit_of_measurement=UnitOfTime.MINUTES,
+        device_class=SensorDeviceClass.DURATION,
+        state_class=SensorStateClass.MEASUREMENT,
+        element="battery_minutes",
+    ),
+    OwletSensorEntityDescription(
+        key="signalstrength",
+        translation_key="signalstrength",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+        element="signal_strength",
+    ),
+    OwletSensorEntityDescription(
+        key="skintemp",
+        translation_key="skintemp",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        state_class=SensorStateClass.MEASUREMENT,
+        element="skin_temperature",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the owlet sensors from config entry."""
+
+    coordinators: list[OwletCoordinator] = list(
+        hass.data[DOMAIN][config_entry.entry_id].values()
+    )
+
+    async_add_entities(
+        OwletSensor(coordinator, sensor)
+        for coordinator in coordinators
+        for sensor in SENSORS
+    )
+
+    async_add_entities(
+        OwletSleepStateSensor(coordinator) for coordinator in coordinators
+    )
+
+
+class OwletSensor(OwletBaseEntity, SensorEntity):
+    """Representation of an Owlet sensor."""
+
+    def __init__(
+        self,
+        coordinator: OwletCoordinator,
+        sensor_description: OwletSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description: OwletSensorEntityDescription = sensor_description
+        self._attr_unique_id = (
+            f"{self.sock.serial}-{self.entity_description.translation_key}"
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return sensor value."""
+
+        if (
+            self.entity_description.element
+            in [
+                "heart_rate",
+                "battery_minutes",
+                "oxygen_saturation",
+                "skin_temperature",
+                "oxygen_10_av",
+            ]
+            and self.sock.properties["charging"]
+        ):
+            return None
+
+        properties = self.sock.properties
+
+        return properties[self.entity_description.element]
+
+
+class OwletSleepStateSensor(OwletBaseEntity, SensorEntity):
```

Comment:
Not sure why this one is needed and not part of the existing entity descriptions?